modules/light-clients/08-wasm/keeper/keeper.go

```diff
@@ -98,6 +98,11 @@ func (k Keeper) storeWasmCode(ctx sdk.Context, code []byte) ([]byte, error) {
 		return nil, errorsmod.Wrap(err, "failed to store contract")
 	}
 
+	// pin the code in the vm
+	if err := k.wasmVM.Pin(codeHash); err != nil {
+		return nil, errorsmod.Wrap(types.ErrPinContractFailed, err.Error())
```

Comment:
Sorry, I know I wrote this in the issue, but now that I look at it, I think it's better if we do (because it's more consistent with error wrapping standards in ibc-go):

```suggestion
		return nil, errorsmod.Wrap(err, "pinning contract failed")
```

And the we don't need to `ErrPinContractFailed`.

Reply:
Resolved in e18cd55eadd98b2f8f50be8063d638fece987967

Reply:
this to pin the contract to the vm LRU cache, correct? nit from me, but maybe we could adjust the in-line code comment and error msg to reflect that?

maybe something like: `errorsmod.Wrap(err, "failed to pin contract to vm cache")`

could be good to check if the `err` returned from `wasmvm.Pin()` includes the `codeHash` or if we should include that also as an `Wrapf()` arg

Reply:
https://github.com/CosmWasm/wasmvm/blob/main/internal/api/lib.go#L105-L114

I reckon best to wrapf the code hash